semantic/parse.go

```diff
@@ -0,0 +1,58 @@
+package semantic
```

Comment:
FYI @G-Rath I looked into adding this lint to golanci, and it is surprisingly difficult because it is disabled by default. You need to enable revive, and then disable the default exclusions listed here: https://golangci-lint.run/usage/false-positives/#default-exclusions, which introduces a large number of linting issues.

Reply:
hmm I actually already looked into that and didn't find the revive rule for this actually triggered and just assumed it wasn't quite what we want, but maybe I just mucked up the configuring of revive.

Either way, enabling revive is something else to explore which I've shelved for once we've gotten through more of the other linters 😅